MvvmCross/MvvmCross
# Move Setup to MvxAndroidApplication +semver:breaking

### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Bug fix/feature

### :arrow_heading_down: What is the current behavior?
Today we try to ensure MvvmCross is initialized in every Activity/Service/BroadcastReceiver/Fragment. This is really unnecessary as we can simply rely on Android Application for our setup.

Android Application, will _always_ run before any Service or Activity is started. This is similar to iOS and other platforms.

### :new: What is the new behavior (if this is a feature change)?

`MvxSplashScreenActivity` has been renamed to `MvxStartActivity`.

MvvmCross is now only initialized in MvxAndroidApplications.

All consumers of MvvmCross on Android will _need_ to have a Application class looking something like this:

```csharp
[Application]
public class MainApplication : MvxAndroidApplication<Setup, App>
{
    public MainApplication(IntPtr javaReference, JniHandleOwnership transfer) : base(javaReference, transfer)
    {
    }
}
```

The major breaking change here is that if you opt to **not** have your `MainLauncher` Activity inherit from `MvxStartActivity` the registered start `ViewModel` in your `MvxApplication` class will be ignored!

If you have a `MvxStartActivity` the startup behavior will continue as it was previously and you can have special logic in your `MvxApplication` to conditionally show a specific `ViewModel` after the splash screen.

### :boom: Does this PR introduce a breaking change?
Yes!

### :bug: Recommendations for testing
Test various scenarios

- App with and without MvxStartActivity

- Set background limit or kill Activities when they are backgrounded using Developer Settings
  - Ensure App behaves correctly and MvvmCross is correctly initialized in when App is killed in background

### :memo: Links to relevant issues/docs
Supersedes #3376 
#3854

### :thinking: Checklist before submitting

- [x] All projects build
- [x] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contributing/mvvmcross-docs-style-guide))
- [x] Rebased onto current develop
